CHANGELOG.unreleased.md

```diff
@@ -1167,6 +1167,31 @@ new Map([
 ]);
 ```
 
+#### TypeScript: Keep semi for a class property before index signature when no-semi is enabled ([#6728] by [@sosukesuzuki])
+
+Attempting to format the stable ver output again results in a syntax error.
+
+<!-- prettier-ignore -->
+```ts
+// Input
+export class User {
+  id: number;
+  [key: string]: any
+}
+
+// Output (Prettier stable)
+export class User {
+  id: number
+  [key: string]: any
+}
```

Comment:
This doesn't break without the semicolon. This does (an initializer is needed):
```ts
export class User {
  id: number = 2;
  [key: string]: any
}
```

Reply:
Thank you, I misunderstood. I fixed it!